stubs/psycopg2/psycopg2/_psycopg.pyi

```diff
@@ -120,19 +122,27 @@ class cursor:
     def cast(self, oid: int, s: str | bytes, /) -> Any: ...
     def close(self) -> None: ...
     def copy_expert(
-        self, sql: str | bytes | Composable, file: _SupportsReadAndReadline | SupportsWrite[str], size: int = 8192
+        self,
+        sql: str | bytes | Composable,
+        file: _SupportsReadAndReadline[bytes] | _SupportsReadAndReadline[str] | SupportsWrite[bytes] | TextIO,
```

Comment:
`_SupportsReadAndReadline[str]` is now redundant with `TextIO`:

```suggestion
        file: _SupportsReadAndReadline[bytes] | SupportsWrite[bytes] | TextIO,
```

Reply:
This seems to be backwards: `TextIO` is redundant with `_SupportsReadAndReadline`. Many "file-like objects" don't have all the things required by `TextIO`.